### ISSUE TITLE
"Go to Implementation" says "No implementation found"

### ISSUE THREAD

**Issue by makkes** [microsoft/vscode-go#2862]
**Describe the bug**

I have this project open in VS Code: https://github.com/makkes/auth-service/ 

Putting the cursor onto an interface in one of the files of the project and then pressing Cmd+Shift+P (on macOS) and selecting "Go to implementation" yields the popup message "No implementation found"

**VSCode settings**
`Preferences: Open Settings (JSON)`
```json
// Settings for VS Code
{
[...]
  "go.useLanguageServer": true,
  "[go]": {
    "editor.codeActionsOnSave": {
      "source.organizeImports": true
    }
[...]
}
```

**Expected behavior**
VS Code should offer me a list of the existing implementations (which definitely are in the project).

**Screenshots**
![Screen-Recording-2019-10-25-at-13 34 17](https://user-images.githubusercontent.com/387444/67568703-349ffd00-f72d-11e9-8b4d-c5bb8600f8f1.gif)

**System information:**
 - OS: macOS 10.14
 - Extension Version: 0.11.7
 - VS Code Version: 1.39.2

**Comment by makkes** [comment]
sorry for the bad quality of the screen recording

**Comment by stamblerre** [comment]
The language server only just added the "Go to implementation" feature, so you will need to use `gopls` at master to have that command work.

**Comment by makkes** [comment]
Thanks, @stamblerre. However, I checked out the repo from https://go.googlesource.com/tools and did a `go install`, restarted VS Code but am still getting the same message.

**Comment by stamblerre** [comment]
I actually just noticed a small bug in the extension that prevents it from working. It will be fixed after https://github.com/microsoft/vscode-go/pull/2876 is merged.

**Comment by ramya-rao-a** [comment]
Please try the [latest beta version](https://github.com/microsoft/vscode-go/wiki/Use-the-beta-version-of-the-latest-Go-extension) of this extension has the fix for this

**Comment by makkes** [comment]
@ramya-rao-a @stamblerre I have toe Go extension v0.11.8-beta.4 now but I'm still getting the same error, even using `gopls` from the master branch.

**Comment by makkes** [comment]
Adding some more context; here's the output from the gopls output:

```
[Info  - 12:00:47 PM] Connection to server got closed. Server will restart.
[Info  - 12:00:48 PM] 2019/11/04 12:00:48 Build info
----------
golang.org/x/tools/gopls v0.1.6
    golang.org/x/tools/gopls@(devel)
    github.com/BurntSushi/toml@v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
    github.com/sergi/go-diff@v1.0.0 h1:Kpca3qRNrduNnOQeazBd0ysaKrUJiIuISHxogkT9RPQ=
    golang.org/x/sync@v0.0.0-20190423024810-112230192c58 h1:8gQV6CLnAEikrhgkHFbMAEhagSSnXWGV915qUMm9mrU=
    golang.org/x/tools@v0.0.0-20191017151554-a3bc800455d5 => ../
    golang.org/x/xerrors@v0.0.0-20190717185122-a985d3407aa7 h1:9zdDQZ7Thm29KFXgAX/+yaf3eVbP7djjWp/dXAppNCc=
    honnef.co/go/tools@v0.0.1-2019.2.3 h1:3JgtbtFHMiCmsznwGVTUWbgGov+pVqnlf1dEJTNAXeM=

Go info
-------
go version go1.13.3 darwin/amd64

GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/mwerner/Library/Caches/go-build"
GOENV="/Users/mwerner/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/mwerner/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go/1.13.3/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.13.3/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/mwerner/dev/personal/services.makk.es/auth-service/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/hb/86f25m_16fgc76qkh9dbjryh0000gp/T/go-build524867019=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

**Comment by ramya-rao-a** [comment]
@stamblerre ?

**Comment by stamblerre** [comment]
@makkes: Can you add the full debug log details? Looks like the stack trace from the crash is cut off there.

**Comment by makkes** [comment]
Here's the complete log: https://filebin.net/rpbqx35insw63aqe/gopls.log?t=7k00k42z

**Comment by matloob** [comment]
The implementation support that was submitted only supported implementations within the same package. Is the implementation of the interface in the same package? If it is not, then I'll mark this as a duplicate of https://github.com/golang/go/issues/32973, and we can follow along there.

If the implementation is in the same package, this is a bug, and I think we'll need more concrete reproduction information.

We need to add support to the LSP for loading the whole project before we can properly support implementations in the same project. I'm starting that work in https://go-review.googlesource.com/c/tools/+/204822.

**Comment by matloob** [comment]
Okay, I'll close this bug in favor of golang/go#32973. Let's continue the discussion there.

Thanks!

**Comment by ramya-rao-a** [comment]
Closing in favor of the upstream issue

Thanks @matloob

**Comment by makkes** [comment]
Thanks @ramya-rao-a @stamblerre @matloob for your support!
